sprockets/mixins/cors.py

```diff
@@ -1,5 +1,143 @@
+
 """
-version_info = (0, 0, 0)
+
+version_info = (0, 1, 0)
 __version__ = '.'.join(str(v) for v in version_info)
+
+SIMPLE_REQUEST_HEADERS = frozenset(('accept', 'accept-language',
+                                    'content-language'))
+"""
+Request headers that get special treatment per CORS.
+
+These are considered "simple headers" and are always acceptable to
+send in CORS-enabled requests.
+
+"""
+
+
+class CORSSettings(object):
+    """
+    Configures the CORS behavior.
+
+    .. attribute:: allowed_methods
+
+       The :class:`set` of CORS accepted HTTP methods.  This controls
+       the :mailheader:`Access-Control-Allow-Methods` response header.
+
+    .. attribute:: allowed_origins
+
+       The :class:`set` of origins that are allowed for the endpoint.
+       This controls the :mailheader:`Access-Control-Allow-Origin`
+       response header.  If the requested origin is in this set, then
+       the origin is allowed; otherwise, a :http:statuscode:`403` is returned.
+
+    .. attribute:: credentials_supported
+
+       Should the mix-in generate the
+       :mailheader:`Access-Control-Allow-Credentials` header in the
+       response.
+
+    .. attribute:: request_headers
+
+       A :class:`set` of header names that are acceptable in cross-origin
+       requests.  Headers added to this set **MUST** be lower-cased before
+       adding them to the set.
+
+    """
+    def __init__(self):
+        self.allowed_methods = set()
+        self.allowed_origins = set()
+        self.credentials_supported = False
+        self.request_headers = set(header.lower()
+                                   for header in SIMPLE_REQUEST_HEADERS)
+
+
+class CORSMixin(object):
+    """
+    Mix this in over a :class:`tornado.web.RequestHandler` for CORS support.
+
+    .. attribute:: cors
+
+       A :class:`.CORSSettings` instance that controls the behavior of
+       the mix-in.
+
+    """
+
+    def initialize(self, **kwargs):
+        self.cors = CORSSettings()
+        self.cors.allowed_origins.update(self.settings.get('cors_origins', []))
+        super(CORSMixin, self).initialize(**kwargs)
+
+    def prepare(self):
+        super(CORSMixin, self).prepare()
+        if not self._finished and self.request.method != 'OPTIONS':
+            origin = self.request.headers.get('Origin')
+            if origin in self.cors.allowed_origins:
+                self.set_header('Access-Control-Allow-Origin', origin)
+                if self.cors.credentials_supported:
+                    self.set_header('Access-Control-Allow-Credentials', 'true')
+
+    def options(self):
+        """
+        Respond to an :http:method:`OPTIONS` request.
+
+        This method relies on :attr:`self.SUPPORTED_METHODS` for the
+        content of the :http:header:`Allow` response header.  The CORS
+        specific headers are generated based on the :attr:`.cors`
+        attribute.
+
+        """
+        self.set_header('Allow', ','.join(self.SUPPORTED_METHODS))
```

Comment:
`tornado.web.RequestHandler` defines it (https://github.com/tornadoweb/tornado/blob/master/tornado/web.py#L154-L155).  I added a pile of tox testing (9b895b0) to make sure that this is safe across versions as well.